server/src/main/java/org/jboss/as/server/moduleservice/ServiceModuleLoader.java

```diff
@@ -64,32 +65,29 @@
      * @author Stuart Douglas
      *
      */
-    private class ModuleSpecLoadListener extends AbstractServiceListener<ModuleDefinition> {
+    private class ModuleSpecLoadListener implements LifecycleListener {
 
+        private final CountDownLatch latch;
```

Comment:
This particular usage of CDL could have just been a simple volatile boolean flag coupled with synchronization for the await logic.  I'd consider CDL usage where there's only ever one count to be overkill.